apps/cyberstorm-remix/cyberstorm/utils/requestCache.ts

```diff
@@ -0,0 +1,69 @@
+import isEqual from "lodash/isEqual";
+
+const restrictedNames = new Set(["", "anonymous", "default"]);
+
+type CacheEntry = {
+  funcName: string;
+  inputs: unknown[];
+  promise: Promise<unknown>;
+};
+
+const requestScopedCaches = new WeakMap<Request, CacheEntry[]>();
+function getCache(request: Request) {
```

Comment:
Nitpick: I absolutely hate that there's no empty line between the variable and the function definition.